### PR TITLE
try lint avoid_unused_constructor_parameters

### PR DESCRIPTION
**Pull request by a14n**

### CODE REVIEW

File: packages/flutter/lib/src/foundation/diagnostics.dart [hunk 1]
```diff
@@ -1955,7 +1955,6 @@ class DiagnosticableNode<T extends Diagnosticable> extends DiagnosticsNode {
     String name,
     @required this.value,
     @required DiagnosticsTreeStyle style,
-    String emptyBodyDescription,
```
Comment (Hixie) [inline packages/flutter/lib/src/foundation/diagnostics.dart]: cc @jacob314

File: packages/flutter/lib/src/rendering/editable.dart [hunk 2]
```diff
@@ -127,6 +127,7 @@ class RenderEditable extends RenderBox {
        _showCursor = showCursor ?? new ValueNotifier<bool>(false),
        _hasFocus = hasFocus ?? false,
        _maxLines = maxLines,
+       _selectionColor = selectionColor,
```
Comment (Hixie) [inline packages/flutter/lib/src/rendering/editable.dart]: cc @HansMuller

**Comment by Hixie** [comment]
LGTM

File: packages/flutter/lib/src/foundation/diagnostics.dart (hunk 1 above)
Comment (jacob314) [reply packages/flutter/lib/src/foundation/diagnostics.dart]: nice! the lint caught some dead code left over after refactoring.

**Pull request merged**